lens/lotusrepo/repo.go

```diff
@@ -0,0 +1,63 @@
+package lotusrepo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/sentinel-visor/lens"
+	"github.com/urfave/cli/v2"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/vm"
+	"github.com/filecoin-project/lotus/node/"
+	"github.com/filecoin-project/lotus/node/impl"
+	"github.com/filecoin-project/lotus/node/repo"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+type RepoAPI struct {
+	api.FullNode
+	context.Context
+}
+
+func (ra *RepoAPI) ComputeGasOutputs(gasUsed, gasLimit int64, baseFee, feeCap, gasPremium abi.TokenAmount) vm.GasOutputs {
+	return vm.ComputeGasOutputs(gasUsed, gasLimit, baseFee, feeCap, gasPremium)
+}
+
+func (ra *RepoAPI) Store() adt.Store {
+	i, ok := ra.FullNode.(*impl.FullNodeAPI)
+	if !ok {
+		return nil
+	}
+	store := i.ChainAPI.Chain.Blockstore()
+	cs := cbor.NewCborStore(store)
+	adtStore := adt.WrapStore(ra.Context, cs)
+	return adtStore
```

Comment:
commentary: At some point we will want caching here, this looks like an easy options: https://github.com/filecoin-project/lotus/blob/master/lib/cachebs/cachebs.go#L22